### PR TITLE
#34935 Execution time formatting

### LINKED ISSUE #21487: Execution log: incorrect sorting by execution time
### Description

Sorting by execution time in execution log is incorrect. It's order like text, but values are numeric.

### DBeaver Version

Community Edition 23.2.2

### Operating System

Windows 10 x64

### Database and driver

PostgreSQL 15
PostgreSQL JDBC Driver 42.5.2

### Steps to reproduce

1. Run several queries.
2. Open the execution log window.
3. Sort by execution time.

### Additional context

_No response_

### LINKED ISSUE #34935: Feature Request: Duration in Minutes
**Is your feature request related to a problem? Please describe.**
Long running scripts show duration in milliseconds, which is not helpful. No one uses milliseconds for anything in the real world. When you go for a run, or microwave something, it's either minutes or seconds. Seeing 514,926 ms run time isn't helpful, because then you have to export to excel, divide that out to get seconds, divide again to get minutes. 

**Describe the solution you'd like**
Simple option in the preferences/settings to identify how long each step of an SQL query is taking. Ex: 15 mins 45 seconds

### PR DESCRIPTION
**Pull request by ShadelessFox**
image

### CODE REVIEW

**Review by E1izabeth — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/TableColumnSortListener.java [hunk 1]
```diff
@@ -78,47 +84,60 @@ private static void sortTable(Table table, Comparator<TableItem> comparator)
...
+        }
 
-    private static Double getNumberFromString(String str) {
-        if (str.isEmpty()) return null;
-        if (!Character.isDigit(str.charAt(0))) return null;
-        int numLength = 1;
-        for (; numLength < str.length(); numLength++) {
-            if (Character.isWhitespace(str.charAt(numLength))) {
-                break;
+        private static Double getNumberFromString(String str) {
```
Comment (ivgag) [inline plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/TableColumnSortListener.java:127]:
```suggestion
        private static Double getNumberFromString(@NotNull String str) {
```

File: plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/TableColumnSortListener.java [hunk 2]
```diff
@@ -78,47 +84,60 @@ private static void sortTable(Table table, Comparator<TableItem> comparator)
...
+    private static class DefaultComparator implements Comparator<TableItem> {
+        private final Collator collator = Collator.getInstance(Locale.getDefault());
+        private final int columnIndex;
+
+        public DefaultComparator(int columnIndex) {
+            this.columnIndex = columnIndex;
+        }
+
+        @Override
+        public int compare(TableItem o1, TableItem o2) {
```
Comment (ivgag) [inline plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/TableColumnSortListener.java:110]:
```suggestion
        public int compare(@NotNull TableItem o1, @NotNull TableItem o2) {
```

**Review by ivgag — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/TableColumnSortListener.java (hunk 1 above)
Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/TableColumnSortListener.java:127]: I've intentionally left this code untouched

**Comment by uslss** [comment]
verified

**Pull request merged**